test/pseudo-tty/no_dropped_stdio.js

```diff
@@ -12,4 +12,13 @@ let out = ('o'.repeat(48) + '\n').repeat(20);
 out += 'o'.repeat(24) + 'O';
 
 process.stdout.write(out);
-process.exit(0);
+
+// In AIX, the child exits even before the python parent
+// can setup the readloop. Provide a reasonable delay.
+if (common.isAix) {
+  setTimeout(function() {
+    process.exit(0);
+  }, 200);
```

Comment:
> I am not sure I understand why this would break the purpose of the test - can you please explain?

What happens in the test is that the Node process writes a chunk of data that is slightly too large to fit into the OS buffer for the TTY (specifically for the OS X value here); for regular streams, this would mean that the data can only partly be written synchronously, and some of the data will need to be written asynchronously.

For TTYs we avoid this by turning them into blocking streams, so all data is written synchronously, and this is what the test here is checking.

That’s why the `process.exit(0)` has to happen in exactly the same tick; otherwise libuv can do asynchronous I/O before the process exits.

> The issue is: there is a race condition between child's exit and parent's read logic

Yeah, it’s odd… I don’t know much about AIX but intermingling the I/O logic and the process control logic seems weird to me.